File: blackbelt/bankAccount.cpp
#include "bankAccount.h"
#include "savingsAccount.h"
#include <iostream>
#include <string>

BankAccount::BankAccount() {
	BankAccount::savingsAccount = new SavingsAccount();
}

BankAccount::BankAccount(std::string firstName, std::string lastName, std::string accountNumber, int accountPIN, bool isAdmin, SavingsAccount* savingsAccount, double balance) {
	BankAccount::firstName = firstName;
	BankAccount::lastName = lastName;
	BankAccount::accountNumber = accountNumber;
	BankAccount::accountPIN = accountPIN;
	BankAccount::isAdmin = isAdmin;
	BankAccount::savingsAccount = savingsAccount;
	BankAccount::balance = balance;
}

BankAccount::~BankAccount() {
	delete savingsAccount;
}




//getters
std::string BankAccount::getFirstName() {
	return BankAccount::firstName;
}

std::string BankAccount::getLastName() {
	return BankAccount::lastName;
}

std::string BankAccount::getAccountNumber() {
	return BankAccount::accountNumber;
}

int BankAccount::getAccountPIN() {
	return BankAccount::accountPIN;
}

bool BankAccount::getIsAdmin() {
	return BankAccount::isAdmin;
}

SavingsAccount BankAccount::getSavingsAccount() {
	return *BankAccount::savingsAccount;
}

double BankAccount::getBalance() {
	return BankAccount::balance;	
}



//setters
void BankAccount::setFirstName(std::string firstName) {

	BankAccount::firstName = firstName;

}

void BankAccount::setLastName(std::string lastName) {

	BankAccount::lastName = lastName;

}

void BankAccount::setAccountNumber(std::string accountNumber) {

	BankAccount::accountNumber = accountNumber;

}

void BankAccount::setAccountPIN(int accountPIN) {

	BankAccount::accountPIN = accountPIN;

}

void BankAccount::setIsAdmin(bool isAdmin) {

	BankAccount::isAdmin = isAdmin;

}

void BankAccount::setSavingsAccount(SavingsAccount* savingsAccount) {

	BankAccount::savingsAccount = savingsAccount;

}

void BankAccount::setBalance(double balance) {

	BankAccount::balance = balance;

}


void BankAccount::deposit(double balance) {
	BankAccount::balance = BankAccount::balance + balance;
}


void BankAccount::withdraw(double balance) {

	if(BankAccount::balance < balance) {

		std::cerr << "Insufficient funds!" << std::endl << std::endl;

	} else {

		BankAccount::balance = BankAccount::balance - balance;

	}

}

File: base/BankAccount.java
public class BankAccount {

	private String firstName;
	private String lastName;
	private String accountNumber;
	private int accountPIN;
	private boolean isAdmin;
	private SavingsAccount savingsAccount;
	private double amount;

	public BankAccount(String firstName, String lastName, String accountNumber, int accountPIN, boolean isAdmin, SavingsAccount savingsAccount, double amount) {

		super();
		this.firstName = firstName;
		this.lastName = lastName;
		this.accountNumber = accountNumber;
		this.accountPIN = accountPIN;
		this.isAdmin = isAdmin;
		this.savingsAccount = savingsAccount;
		this.amount = amount;

	}


	//getters

	public String getFirstName() {

		return firstName;

	}

	public String getLastName() {

		return lastName;

	}

	public String getAccountNumber() {

		return accountNumber;

	}

	public int getAccountPIN() {

		return accountPIN;

	}

	public boolean getIsAdmin() {

		return isAdmin;

	}

	public SavingsAccount getSavingsAccount() {

		return savingsAccount;

	}

	public double getAmount() {

		return amount;

	}


	//setters

	public void setFirstName(String firstName) {

		this.firstName = firstName;

	}

	
	public void setLastName(String lastName) {

		this.lastName = lastName;

	}


	public void setAccountNumber(String accountNumber) {

		this.accountNumber = accountNumber;

	}

	
	public void setAccountPIN(int accountPIN) {

		this.accountPIN = accountPIN;

	}


	public void setIsAdmin(boolean isAdmin) {

		this.isAdmin = isAdmin;

	}


	public void setAmount(double amount) {

		this.amount = amount;

	}


	//functions


	public void deposit(double amount) {

		this.amount = this.amount + amount;

	}


	public void withdraw(double amount) {

		if(this.amount < amount) {

			System.out.println("Insuficient funds for this transaction.");

		} else {

			this.amount = this.amount - amount;

		}

	}


}